examples/realworld/ext/user/actions.js

```diff
@@ -1,20 +1,32 @@
 import HttpError from '@wasp/core/HttpError.js'
+import AuthError from '@wasp/core/AuthError.js'
+import { isPrismaError, prismaErrorToHttpError } from '@wasp/utils.js'
 
 export const updateUser = async ({ email, username, bio, profilePictureUrl, newPassword }, context) => {
   if (!context.user) { throw new HttpError(403) }
 
   // TODO: Nicer error handling! Right now everything is returned as 500 while it could be instead
   //   useful error message about username being taken / not unique, and other validation errors.
-  await context.entities.User.update({
-    where: { id: context.user.id },
-    data: {
-      email,
-      username,
-      bio,
-      profilePictureUrl,
-      ...(newPassword ? { password: newPassword } : {})
+  try {
+    await context.entities.User.update({
+      where: { id: context.user.id },
+      data: {
+        email,
+        username,
+        bio,
+        profilePictureUrl,
+        ...(newPassword ? { password: newPassword } : {})
+      }
+    })
+  } catch (e) {
+    if (e instanceof AuthError) {
+      throw new HttpError(422, 'Validation failed', { message: e.message })
+    } else if (isPrismaError(e)) {
+      throw prismaErrorToHttpError(e)
+    } else {
+      throw new HttpError(500)
```

Comment:
I think here we can even just rethrow the old error -> there will be part of Wasp that will then ensure to convert it into HttpError 500.